src/librustc/lib.rs

```diff
@@ -68,6 +68,7 @@ extern crate serialize as rustc_serialize; // used by deriving
 #[cfg(test)]
 extern crate test;
 
+#[allow(private_in_public)]
```

Comment:
Is `pub extern crate` very buggy? I think it can be used here instead of suppressing the warning. It will report the "`pub extern crate` does not work as expected" warning during `stage0`, but that's ok.